app/src/org/commcare/activities/FormEntryActivity.java

```diff
@@ -632,7 +632,7 @@ public Object onRetainCustomNonConfigurationInstance() {
             return mSaveToDiskTask;
 
         // mFormEntryController is static so we don't need to pass it.
-        if (mFormController != null && currentPromptIsQuestion()) {
+        if (mFormController != null && currentPromptIsQuestion() && uiController.questionsView != null) {
             saveAnswersForCurrentScreen(FormEntryConstants.DO_NOT_EVALUATE_CONSTRAINTS);
```

Comment:
Hmm, isn't this call pretty important? Is it ok for this not to be called, or can we be sure that this will be called again once `uiController.questionsView` isn't null?

Reply:
My reasoning was that if `questionsView` is null, then there is no current screen with answers that needs to be saved

Reply:
Ah OK understood, that seems totally sound.